src/KeymapsDialogManager.ts

```diff
@@ -0,0 +1,87 @@
+import { Editor, PluginOptions } from 'grapesjs'
+import { KeymapsDialog } from './KeymapsDialog'
+
+interface KeymapsInfo {
+    name: string,
+    keys: string[]
+}
+
+interface KeymapsRegistry {
+    [key: string]: KeymapsInfo[]
+}
+
+/**
+ * Capitalizes the first letter of a string and lowercases the rest.
+ * @param str The string to capitalize.
+ */
+export function titleCase(str: string): string {
+    return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase()
+}
+
+/**
+ * Manages the state of the keymaps help dialog.
+ */
+export class KeymapsDialogManager {
+    /**
+     * Whether the keymaps registry is dirty and needs to be updated.
+     */
+    isDirty: boolean
+    editor: Editor
+    dialog: KeymapsDialog
+    keymapsRegistry: KeymapsRegistry
+
+    constructor(editor: Editor, opts: PluginOptions) {
+        this.isDirty = true
+        this.editor = editor
+        this.dialog = new KeymapsDialog(this, opts)
+        this.keymapsRegistry = {}
+
+        // Asks for a registry update next time the keymaps are modified
+        editor.on('keymap:add keymap:remove', () => {
+            this.isDirty = true
```

Comment:
Maybe update the UI?

Reply:
i think it is simpler and better to just refresh the UI
if you want to be sure it is not called too often you can use a debounce pattern (but maybe overkill here)

also you should listen the editor events only when the dialog is open or at least do nothing if it is closed

Reply:
Understood, it may be overkill too. Let's do this!